### DISCUSSION TITLE
Is there a way to force restart some quests ?

### DISCUSSION THREAD

**Discussion by xubitus** [streetcomplete/StreetComplete#3946] [Q&A] [answered]
I have been updating my city regarding bike lanes, mainly indicating that there are none or very few.
Recently, the city started painting new signs on the roads and adding "shared" streets and more of the kind.
I would love to update the informations on this but since I tagged everything already, the quests are not poping up anymore.

Is there a way to force the app to ask again the same quests so as to update the values ?

**Comment by FloEdelmann** [comment, marked as answer]
You can set the resurvey interval to ask more often in the settings (see also https://github.com/streetcomplete/StreetComplete/issues/3909#issuecomment-1076403764). The default resurvey interval cycleway quest is currently 4 years (see also https://wiki.openstreetmap.org/wiki/StreetComplete/Quests), so that would allow you to re-answer the quest for all roads that haven't been touched for at least 2 years.

**Reply by xubitus** [reply]
OK I'll try that but I think most of my updates have been done during the last 2 years. The deployment is still ongoing in the city so I'll wait a bit anyway for it to be stabilized.
I guess mass changes like that are not very common and it seems logical there are dedicated features for that.

**Reply by Helium314** [reply]
Will this resurvey really happen? Looking at https://github.com/streetcomplete/StreetComplete/blob/23b70c6ff32160fda82e33e5ac706607f55e1a53/app/src/main/java/de/westnordost/streetcomplete/quests/cycleway/AddCycleway.kt#L369 I would expect it does not work if there is no `cycleway` tag.
SC often tags `cycleway:both`, or `cycleway:left` and `cycleway:right`, without a `cycleway` tag.

**Reply by matkoniecz** [reply]
Good point, it looks like a bug to me (I will put it on my TODO list, as I would need to investigate it a bit to be sure)

note for future me: https://github.com/streetcomplete/StreetComplete/blob/master/app/src/main/java/de/westnordost/streetcomplete/quests/cycleway/AddCycleway.kt#L369 can be checked whether code was modified

**Reply by peternewman** [reply]
It looks like it's even more true for recycling which only ever tags `recycling:`:
https://github.com/streetcomplete/StreetComplete/blob/0e900879c915be0d2ab0ee4df234b3a338447711/app/src/main/java/de/westnordost/streetcomplete/quests/recycling_material/AddRecyclingContainerMaterials.kt#L124

Indeed this seems to confirm it, it only offers a check existence quest:
https://www.openstreetmap.org/node/279651296

**Reply by arrival-spring** [reply]
These quests add `check_date:recycling` and `check_date:cycleway` if nothing has changed, so that is the tag being checked for.

The recycling one at least definitely works, I have resurvey set to more often and I remember recently resurveying a container after only one year.

**Reply by peternewman** [reply]
> Indeed this seems to confirm it, it only offers a check existence quest: https://www.openstreetmap.org/node/279651296

Turns out my example is broken by recycling:small_appliances/recycling:small_electrical_appliances .

**Reply by peternewman** [reply]
So the answer as found in #3964 and #3965 is that if it doesn't match the tag its still a match if it's old enough, and it will then fall back to the general selection logic.

**Comment by riQQ** [comment]
There's also a feature request for forcing a resurvey: #3918

**Reply by FloEdelmann** [reply]
See also #3644, #2963, #3091, #2477, #3083, #3237, ...
